text/chapter2.md

```diff
@@ -90,7 +86,7 @@ And check our work by running `spago test`:
 
```

Comment:
Up on lines 72-75, there's some temporary code being introduced (a faulty version of `diagonal`), which isn't kept in the final solution.  I'm assuming it's best to just leave that code snippet written directly into the .md file instead of putting that in one of the .purs files and anchoring it?